File: Backend/shop/shop-typescript-nestjs-npm-typeorm-postgresql/src/auth/auth.controller.ts
import { Body, Controller, Post } from "@nestjs/common";
import { ApiBearerAuth, ApiOperation, ApiTags } from "@nestjs/swagger";
import { ok } from "../common/api-response";
import { CurrentUser } from "../common/current-user.decorator";
import type { CurrentUserPayload } from "../common/current-user.decorator";
import { Public } from "../common/public.decorator";
import { AuthService } from "./auth.service";

@ApiTags("auth")
@Controller("api/v1/auth")
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  @Public()
  @Post("signup")
  @ApiOperation({ summary: "User signup" })
  async signup(@Body() body: Record<string, unknown>) {
    return ok(await this.authService.signup(body));
  }

  @Public()
  @Post("login")
  @ApiOperation({ summary: "User login" })
  async login(@Body() body: Record<string, unknown>) {
    return ok(await this.authService.login(body));
  }

  @Public()
  @Post("refresh")
  @ApiOperation({ summary: "Refresh user tokens" })
  async refresh(@Body() body: Record<string, unknown>) {
    return ok(await this.authService.refresh(body));
  }

  @Post("logout")
  @ApiBearerAuth()
  @ApiOperation({ summary: "User logout" })
  async logout(@CurrentUser() user: CurrentUserPayload) {
    return ok(await this.authService.logout(user));
  }
}

File: Backend/post/post-java-springboot-gradle-mysql/src/main/java/com/projectbible/post/auth/controller/AuthController.java
package com.projectbible.post.auth.controller;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.*;

import com.projectbible.post.auth.dto.AuthDtos.LoginRequestDto;
import com.projectbible.post.auth.dto.AuthDtos.LoginResponseDto;
import com.projectbible.post.auth.dto.AuthDtos.RefreshRequestDto;
import com.projectbible.post.auth.dto.AuthDtos.SignupRequestDto;
import com.projectbible.post.auth.service.AuthService;
import com.projectbible.post.common.api.ApiResponse;
import com.projectbible.post.common.api.MessageResponse;
import com.projectbible.post.common.security.AuthContext;
import com.projectbible.post.user.dto.UserDtos.UserSummaryDto;

@Tag(name = "auth")
@RestController
@RequestMapping("/api/v1/auth")
public class AuthController {
    private final AuthService authService;

    public AuthController(AuthService authService) {
        this.authService = authService;
    }

    @Operation(summary = "User signup")
    @PostMapping("/signup")
    public ApiResponse<UserSummaryDto> signup(@Valid @RequestBody SignupRequestDto body) {
        return ApiResponse.success(authService.signup(body));
    }

    @Operation(summary = "User login")
    @PostMapping("/login")
    public ApiResponse<LoginResponseDto> login(@Valid @RequestBody LoginRequestDto body) {
        return ApiResponse.success(authService.login(body));
    }

    @Operation(summary = "User refresh")
    @PostMapping("/refresh")
    public ApiResponse<LoginResponseDto> refresh(@Valid @RequestBody RefreshRequestDto body) {
        return ApiResponse.success(authService.refresh(body));
    }

    @Operation(summary = "User logout")
    @PostMapping("/logout")
    public ApiResponse<MessageResponse> logout(HttpServletRequest request) {
        return ApiResponse.success(authService.logout(AuthContext.requireUser(request)));
    }
}
